File: NearbyPlaces/CustomView.m
//
//  CustomView.m
//  NearbyPlaces
//
//  Created by Adrian Kubała on 14.02.2017.
//  Copyright © 2017 Adrian Kubała. All rights reserved.
//

#import "CustomView.h"

@interface CustomView ()

@property (nonnull, nonatomic) UIView *view;

@end

@implementation CustomView

-(instancetype)initWithView:(UIView *)view {
  self = [super init];
  
  if (self) {
    _view = view;
  }
  
  return self;
}

-(void)drawViewByBezierPathWithOffset:(CGFloat)leftBottomCornerOffset color:(UIColor *)color {
  UIBezierPath *bezierPath = [[UIBezierPath alloc] init];
  [bezierPath moveToPoint:self.view.bounds.origin];
  
  CGFloat maxX = CGRectGetMaxX(self.view.bounds);
  CGFloat maxY = CGRectGetMaxY(self.view.bounds);
  CGPoint origin = self.view.bounds.origin;
  
  [bezierPath addLineToPoint:CGPointMake(maxX, origin.y)];
  [bezierPath addLineToPoint:CGPointMake(maxX, maxY)];
  [bezierPath addLineToPoint:CGPointMake(origin.x, maxY - leftBottomCornerOffset)];
  [bezierPath closePath];
  
  [color setFill];
  [bezierPath fill];
  
  [color setStroke];
  [bezierPath stroke];
}

@end

File: NearbyPlaces/CustomView.swift
//
//  CustomView.swift
//  Places
//
//  Created by Adrian on 28.09.2016.
//  Copyright © 2016 Adrian Kubała. All rights reserved.
//

import UIKit

@IBDesignable
class CustomView {
  fileprivate let view: UIView
  
  init(view: UIView) {
    self.view = view
  }
  
  func drawViewByBezierPath(_ leftBottomCornerOffset: CGFloat, with color: UIColor) {
    let bezierPath = UIBezierPath()
    bezierPath.move(to: view.bounds.origin)
    bezierPath.addLine(to: CGPoint(x: view.bounds.maxX, y: view.bounds.origin.y))
    bezierPath.addLine(to: CGPoint(x: view.bounds.maxX, y: view.bounds.maxY))
    bezierPath.addLine(to: CGPoint(x: view.bounds.origin.x, y: view.bounds.maxY - leftBottomCornerOffset))
    bezierPath.close()
    
    color.setFill()
    bezierPath.fill()
    
    color.setStroke()
    bezierPath.stroke()
  }
}
